### PR TITLE
Send status update on first block received.

### PR DESCRIPTION
**Pull request by pvyawaha**
The change adds a condition to send status update when first block is received.

Description
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is formatted using Uncrustify.
- [ ] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/ota_mqtt.c [hunk 1]
```diff
@@ -628,7 +628,7 @@ static uint32_t buildStatusMessageReceiving( char * pMsgBuffer,
     received = numBlocks - pOTAFileCtx->blocksRemaining;
 
     /* Output a status update once in a while. */
```
Comment (ravibhagavandas) [inline source/ota_mqtt.c:630]: Can you update the comment to reflect the changes done?

File: source/ota_mqtt.c [hunk 2]
```diff
@@ -628,7 +628,7 @@ static uint32_t buildStatusMessageReceiving( char * pMsgBuffer,
...
-    if( ( received % otaconfigOTA_UPDATE_STATUS_FREQUENCY ) == 0U )
+    if( ( received == 1 ) || ( ( received % otaconfigOTA_UPDATE_STATUS_FREQUENCY ) == 0U ) )
```
Comment (ravibhagavandas) [inline source/ota_mqtt.c:631]: Will the function be called for each block received? Can we add/update unit tests for the changes?

File: source/ota_mqtt.c (hunk 1 above)
Comment (ActoryOu) [reply source/ota_mqtt.c:630]: Comment updated in the #425.

File: source/ota_mqtt.c (hunk 2 above)
Comment (ActoryOu) [reply source/ota_mqtt.c:631]: Yes, the function is called for every block received. Added a test case in #425.

File: source/ota_mqtt.c (hunk 1 above)
Comment (ravibhagavandas) [reply source/ota_mqtt.c:630]: I see PR is closed. Is the plan to update this PR with the changes?

Comment (ravibhagavandas) [reply source/ota_mqtt.c:630]: I see PR is closed. Is the plan to update this PR with the changes?

File: source/ota_mqtt.c (hunk 2 above)
Comment (ravibhagavandas) [reply source/ota_mqtt.c:631]: I see PR is closed. Is the plan to update this PR with the changes?

Comment (ravibhagavandas) [reply source/ota_mqtt.c:631]: I see PR is closed. Is the plan to update this PR with the changes?

File: source/ota_mqtt.c (hunk 1 above)
Comment (ActoryOu) [reply source/ota_mqtt.c:630]: Sorry for late response, I created a new PR for this at #426.

**Pull request closed without merging**